Analysis/Tasks/PWGHF/taskLcK0sp.cxx

```diff
@@ -0,0 +1,157 @@
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file taskLcK0sp.cxx
+/// \brief LcK0sp analysis task
+///
+/// \author Chiara Zampolli, <Chiara.Zampolli@cern.ch>, CERN
+///
+/// based on taskD0.cxx, taskLc.cxx
+
+#include "Framework/AnalysisTask.h"
+#include "Framework/HistogramRegistry.h"
+#include "AnalysisDataModel/HFSecondaryVertex.h"
+#include "AnalysisDataModel/HFCandidateSelectionTables.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::aod::hf_cand_casc;
+using namespace o2::framework::expressions;
+
+void customize(std::vector<o2::framework::ConfigParamSpec>& workflowOptions)
+{
+  ConfigParamSpec optionDoMC{"doMC", VariantType::Bool, false, {"Fill MC histograms."}};
+  workflowOptions.push_back(optionDoMC);
+}
+
+#include "Framework/runDataProcessing.h"
+
+/// LcK0sp analysis task
+struct TaskLcK0sp {
+  HistogramRegistry registry{
+    "registry",
+    {{"hmass", "cascade candidates;inv. mass (#pi K) (GeV/#it{c}^{2});entries", {HistType::kTH1F, {{500, 0., 5.}}}},
+     {"hptcand", "cascade candidates;candidate #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
+     {"hptv0", "cascade candidates;v0 #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
+     {"hptbach", "cascade candidates;bachelor #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
+     {"hd0bach", "cascade candidates;bachelor DCAxy to prim. vertex (cm);entries", {HistType::kTH1F, {{100, -1., 1.}}}},
+     {"hd0v0pos", "cascade candidates;pos daugh v0 DCAxy to prim. vertex (cm);entries", {HistType::kTH1F, {{100, -1., 1.}}}},
+     {"hd0v0neg", "cascade candidates;neg daugh v0 DCAxy to prim. vertex (cm);entries", {HistType::kTH1F, {{100, -1., 1.}}}},
+     {"hv0CPA", "cascade candidates;v0 cosine of pointing angle;entries", {HistType::kTH1F, {{110, 0.998, 1.1}}}},
+     {"hEta", "cascade candidates;candidate #it{#eta};entries", {HistType::kTH1F, {{100, -2., 2.}}}},
+     {"hselectionstatus", "cascade candidates;selection status;entries", {HistType::kTH1F, {{5, -0.5, 4.5}}}}}};
+
+  Configurable<int> d_selectionFlagLcK0sp{"d_selectionFlagLcK0sp", 1, "Selection Flag for LcK0sp"};
+  Configurable<double> cutEtaCandMax{"cutEtaCandMax", -1., "max. cand. pseudorapidity"};
+
+  Filter filterSelectCandidates = (aod::hf_selcandidate_lc_k0sp::isSelLcK0sp >= d_selectionFlagLcK0sp);
+
+  void process(soa::Filtered<soa::Join<aod::HfCandCascExt, aod::HFSelLcK0spCandidate>> const& candidates)
+  {
+    //Printf("Candidates: %d", candidates.size());
+    for (auto& candidate : candidates) {
+      /*
+      // no such selection for LcK0sp for now - it is the only cascade
+      if (!(candidate.hfflag() & 1 << D0ToPiK)) {
+        continue;
+      }
+      */
+      if (cutEtaCandMax >= 0. && std::abs(candidate.eta()) > cutEtaCandMax) {
+        //Printf("Candidate: eta rejection: %g", candidate.eta());
```

Comment:
I suppose this will become a rapidity cut eventually.

Reply:
Yes. For now, I uniformized with D0 and Lc --> pKpi